helix-view/src/document.rs

```diff
@@ -141,27 +141,37 @@ impl Document {
 
     // TODO: async fn?
     pub fn load(path: PathBuf) -> Result<Self, Error> {
-        use std::{fs::File, io::BufReader};
-
-        let doc = if !path.exists() {
+        let text = if !path.exists() {
             Rope::from("\n")
         } else {
-            let file = File::open(&path).context(format!("unable to open {:?}", path))?;
-            let mut doc = Rope::from_reader(BufReader::new(file))?;
-            // add missing newline at the end of file
-            if doc.byte(doc.len_bytes() - 1) != b'\n' {
-                doc.insert_char(doc.len_chars(), '\n');
-            }
-            doc
+            read_file(&path)?
         };
 
-        let mut doc = Self::new(doc);
+        let mut doc = Self::new(text);
         // set the path and try detecting the language
         doc.set_path(&path)?;
 
         Ok(doc)
     }
 
+    pub fn reload(&mut self) -> Result<(), anyhow::Error> {
+        let path = self
+            .path()
+            .ok_or_else(|| Error::msg("unable to reload document without path"))?;
+        let text = read_file(path)?;
+
+        self.changes = ChangeSet::new(&text);
+        self.history = Cell::new(History::default());
```

Comment:
Why should reload change history?

Reply:
@pickfire I just came to the same conclusion here. In vim `:e!` allows you to undo the read if needed and undo history is preserved.

Reply:
I think we should redo it, it will probably mess with the LSP if we merge it as is.

Reply:
Yeah, I'm working on a proper implementation using `Transaction`. Let's keep it open for now.